inst/rmarkdown/templates/transmissibility/skeleton/skeleton.Rmd

```diff
@@ -376,46 +374,37 @@ discretized Gamma.
 
 ## Results
 
-```{r, eval = params$use_epiparameter}
-si_epiparameter <- epiparameter::epidist(
-  params$epiparameter_pathogen,
-  "serial_interval"
+```{r, eval = params$use_epiparameter_database}
+si_epiparameter <- epiparameter::epidist_db(
+  disease =  params$epiparameter_disease,
+  epi_dist =  "serial_interval",
+  single_epidist = TRUE
 )
-si_params <- si_epiparameter$param
-si_dist <- si_epiparameter$dist
-si_mean <- si_params[[1]]
-si_sd <- si_params[[2]]
+
+si_params <- as.list(get_parameters(si_epiparameter))
+si_dist <- family(si_epiparameter)
+si_mean <- si_epiparameter$summary_stats$mean
+si_sd <- si_epiparameter$summary_stats$sd
 ```
 
-```{r, eval = !params$use_epiparameter}
+```{r, eval = !params$use_epiparameter_database}
 si_mean <- params$si_mean
 si_sd <- params$si_sd
 si_dist <- params$si_dist
 si_cv <- si_sd / si_mean
-si_params <- epitrix::gamma_mucv2shapescale(mu = si_mean, cv = si_cv)
+si_params <- epiparameter::convert_summary_stats_to_params(
+  distribution = "gamma", mean = si_mean, sd = si_sd
+)
 ```
 
 ```{r}
-# do.call() because we want to pass `si_params` names as the name of arguments.
-# This is important if params in si_params are not the first arguments of the
-# q...() function. For example, if a gamma distribution is define by its shape
-# and scale, instead of shape and rate.
-si <- do.call(
-  distcrete,
-  c(
-    list(si_dist,
-      interval = 1,
-      w = 0
-    ),
-    si_params
-  )
-)
-si_x <- seq(1L, to = si$q(0.999), by = 1L)
+si <- epiparameter::discretise(si_epiparameter)
+si_x <- seq(1L, to = si$prob_dist$qf(0.999), by = 1L)
```

Comment:
```suggestion
si <- as.function(si_epiparameter)
si_x <- seq(1L, to = quantile(si_epiparameter, 0.999), by = 1L)
```

Note however that in the case `use_epiparameter_database = FALSE`, the object `si_epiparameter` does not exist.